api/iceberg-service/src/main/java/org/apache/polaris/service/types/NotificationType.java

```diff
@@ -84,9 +83,9 @@ public static Optional<NotificationType> lookupByName(String name) {
       return Optional.empty();
     }
 
-    for (NotificationType NotificationType : NotificationType.values()) {
-      if (name.toUpperCase(Locale.ROOT).equals(NotificationType.name())) {
-        return Optional.of(NotificationType);
+    for (NotificationType notificationType : NotificationType.values()) {
```

Comment:
the `lookupByName` method seems completely unused, so we could also just remove it instead afaict.

same applies to the `displayName` field (whose value is always the same as `name()`)